apps/native-component-list/src/screens/Camera/CameraScreenNextBarcode.tsx

```diff
@@ -0,0 +1,28 @@
+import { Camera } from 'expo-camera/next';
+import { useEffect } from 'react';
+import { View, Button } from 'react-native';
+
+export default function CameraScreenNextBarcode() {
+  useEffect(() => {
+    const subscription = Camera.onModernBarcodeScanned((event) => {
+      console.log(event);
+    });
+
+    return () => subscription.remove();
+  }, []);
+
+  async function launchScanner() {
+    if (Camera.modernBarcodeScannerAvailable) {
+      await Camera.launchModernScanner({
```

Comment:
What is the "modern" scanner?

Reply:
Couldn't come up with a better name at the time 😅 It's [this](https://developer.apple.com/documentation/visionkit/datascannerviewcontroller)